sync/src/chain.rs

```diff
@@ -862,16 +863,12 @@ impl ChainSync {
 				warn!(target:"sync", "Asking {:?} while requesting {:?}", peer.asking, asking);
 			}
 		}
-		match sync.send(peer_id, packet_id, packet) {
-			Err(e) => {
-				debug!(target:"sync", "Error sending request: {:?}", e);
-				sync.disable_peer(peer_id);
-			}
-			Ok(_) => {
-				let mut peer = self.peers.get_mut(&peer_id).unwrap();
-				peer.asking = asking;
-				peer.ask_time = time::precise_time_s();
-			}
+		let mut peer = self.peers.get_mut(&peer_id).unwrap();
+		peer.asking = asking;
```

Comment:
Always set the asking status so that requested blocks would be unmarked as being downloaded on disconnect.